src/components/settings/SettingsLMSTab/ErrorReporting/ContentMetadataTable.jsx

```diff
@@ -1,46 +1,91 @@
-import React, { useEffect, useState } from 'react';
+import _ from 'lodash';
+import React, {
+  useCallback, useMemo, useEffect, useState,
+} from 'react';
 import PropTypes from 'prop-types';
 import {
   DataTable, TextFilter,
 } from '@edx/paragon';
 import { logError } from '@edx/frontend-platform/logging';
 import LmsApiService from '../../../../data/services/LmsApiService';
 import DownloadCsvButton from './DownloadCsvButton';
-import { getSyncStatus, getSyncTime } from './utils';
+import { getSyncStatus, getSyncTime, createLookup } from './utils';
 
 function ContentMetadataTable({ config, enterpriseCustomerUuid }) {
-  const [data, setData] = useState([]);
+  const [currentPage, setCurrentPage] = useState();
+  const [currentFilters, setCurrentFilters] = useState();
+  const [paginationData, setPaginationData] = useState({
+    itemCount: 0,
+    pageCount: 0,
+    data: [],
+    sortBy: '',
+  });
 
   useEffect(() => {
     const fetchData = async () => {
       const response = await LmsApiService.fetchContentMetadataItemTransmission(
-        enterpriseCustomerUuid, config.channelCode, config.id,
+        enterpriseCustomerUuid, config.channelCode, config.id, currentPage, currentFilters,
       );
       return response;
     };
 
     fetchData()
       .then((response) => {
-        setData(response.data.results);
+        setPaginationData({
+          itemCount: response.data.count,
+          pageCount: response.data.pages_count,
+          data: response.data.results,
+        });
       })
       .catch((err) => {
         logError(err);
       });
-  }, [config.channelCode, config.id, enterpriseCustomerUuid]);
+  }, [config.channelCode, config.id, enterpriseCustomerUuid, currentPage, currentFilters]);
+
+  // Call back function, handles filters and page changes
+  const fetchData = useCallback(
+    (args) => {
+      let newFilters = createLookup(args.filters, (filter) => filter.id, (filter) => filter.value);
+
+      const sortBy = args.sortBy.at(-1);
+      if (!_.isEmpty(sortBy)) {
+        const newSortBys = { sort_by: `${sortBy.desc ? '-' : ''}${sortBy.id}` };
+        newFilters = { ...newFilters, ...newSortBys };
+      }
+      setCurrentFilters(newFilters);
+
+      if (args.pageIndex !== currentPage) {
+        setCurrentPage(args.pageIndex);
+      }
+    },
+    [setCurrentPage, currentPage],
+  );
+
+  const memoisedFetchData = useMemo(() => ((args) => fetchData(args)), [fetchData]);
```

Comment:
nit: isn't it memoized?

Reply:
ahh yes, this is from my desperate debug session, I think it's redundant now